src/porepy/numerics/ad/equation_system.py

```diff
@@ -851,56 +818,49 @@ def _append_dofs(self, variable: pp.ad.Variable) -> None:
             ) + variable.domain.num_nodes * local_dofs.get("nodes", 0)
 
         # Update the global dofs and block numbers
-        self._variable_numbers.update({variable: last_variable_number})
+        self._variable_numbers.update({variable.id: last_variable_number})
         self._variable_num_dofs = np.concatenate(
             [self._variable_num_dofs, np.array([num_dofs], dtype=int)]
         )
 
-        # first optimization of Jacobian structure
-        self._cluster_dofs_gridwise()
-
     def _cluster_dofs_gridwise(self) -> None:
         """Re-arranges the DOFs grid-wise s.t. we obtain grid-blocks in the column sense
         and reduce the matrix bandwidth.
 
         The aim is to impose a more block-diagonal-like structure on the Jacobian where
         blocks in the column sense represent single grids in the following order:
 
-        Note:
-            Off-diagonal blocks will still be present if subdomain-interface variables
-            are defined.
-
         1. For each grid in ``mdg.subdomains``
             1. For each variable defined on that grid
         2. For each grid in ``mdg.interfaces``
             1. For each variable defined on that mortar grid
 
-        The order of variables per grid is given by the order of variable creation
-        (stored as order of keys in ``self.variables``).
-
+        The order of variables per grid is given by the order of variable creation.
         This method is called after each creation of variables and respective DOFs.
-        TODO: Revisit. I think I have broken it by looping over _variables instead of
-        subdomains and interfaces.
 
         """
         # Data stracture for the new order of dofs.
         new_variable_counter: int = 0
-        new_variable_numbers: dict[Variable, int] = dict()
+        new_variable_numbers: dict[int, int] = dict()
         new_block_dofs: list[int] = list()
 
-        for variable in self._variables:
```

Comment:
Grid-wise clustering of variable DOFs was broken in some previous PR. Fixed again

Reply:
It seems like there is no test covering it as of now.

Reply:
I made a placeholder [issue](https://github.com/pmgbergen/porepy/issues/1167).